includes/Admin/Admin_AJAX.php

```diff
@@ -208,9 +209,39 @@ public function run_checks() {
 			wp_send_json_error( $valid_nonce, 403 );
 		}
 
+		$runner = Plugin_Request_Utility::get_runner();
+
+		if ( is_null( $runner ) ) {
+			$runner = new AJAX_Runner();
+		}
+
+		// Make sure we are using the correct runner instance.
+		if ( ! ( $runner instanceof AJAX_Runner ) ) {
+			wp_send_json_error(
+				new WP_Error( 'invalid-runner', __( 'AJAX Runner was not initialized correctly.', 'plugin-check' ) ),
+				500
+			);
+		}
+
+		$checks = isset( $_REQUEST['checks'] ) ? array_filter( $_REQUEST['checks'] ) : array();
```

Comment:
Why is that done differently here from above? I would think we should use the same logic everywhere for this parameter.
```suggestion
		$checks = filter_input( INPUT_POST, 'checks', FILTER_DEFAULT, FILTER_FORCE_ARRAY );
		$checks = is_null( $checks ) ? array() : $checks;
```

Reply:
@felixarntz good catch! Not sure why this was reverted but have updated this to match.